app/controllers/idv/personal_key_controller.rb

```diff
@@ -76,8 +76,7 @@ def generate_personal_key
 
     def in_person_enrollment?
       return false unless IdentityConfig.store.in_person_proofing_enabled
-      # WILLFIX: After LG-6872 and we have enrollment saved, reference enrollment instead.
-      ProofingComponent.find_by(user: current_user)&.document_check == Idp::Constants::Vendors::USPS
+      current_user.pending_in_person_enrollment.exists?
```

Comment:
Note that as of #6635, a user who chooses to verify their address by mail (the "GPO" path) will not have an enrollment created until they successfully enter their GPO code, so there would not be an existing enrollment at this point in the flow.

I think we should either:

1. Keep this as it was before checking the proofing component, and remove the inline "WILLFIX" comment
2. Or, revise the enrollment creation logic so that we _do_ create an enrollment record in our system for a GPO user, but it is not sent to USPS (and maybe kept as `establishing` ?) until the successful code entry.
   - This may be harder because we currently use the record's creation date as the start for the 30 day timer, but that assumes that the record creation and the USPS enrollment happen simultaneously. We may need a separate `expires_at` schema field if we were to pursue this option.

Reply:
I believe this is resolved now that you've updated the logic to create an `establishing` enrollment at the location page

Reply:
Yeah, following the logic, this looks right now, since there'll always be an enrollment. In the GPO case, it'll still be "establishing" at this point, but considering that we're using this to determine whether to send the user to the "ready to verify" page, we'd only want the pending enrollments, not users who have gone through GPO.

I'll mark as resolved.